InciCsvCorrector.java

```diff
@@ -0,0 +1,159 @@
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class for the correction of the csv file containing the INCI ingredients found at
+ * "http://ec.europa.eu/growth/tools-databases/cosing/pdf/COSING_Ingredients-Fragrance%20Inventory_v2.csv".
+ * There are reoccurring formatting errors in the downloaded file that make it unusable (line endings inside the ingredient description),
+ * this class fixes those errors by writing in a new file the fixed text, the paths to the source and end files will be given by the user.
+ * The INCI database is updated quite often so this class can be used every time there is an update to get a usable file.
+ * @author Nicolò Cervo (g3)
+ */
+public class InciCsvCorrector {
+
+    private static final int DATE_LENGTH = 10;  // dd/mm/yyyy
+    private static final int MINIMUM_VALID_LENGTH = 15; // of a line
+
+    public static void main(String[] args)
+    {
+        String[] paths = getFilesPaths();
+        File source = new File(paths[0]);
+        File corrected = new File(paths[1]);
+        boolean needsCorrection = false;
+        ArrayList<Integer> invLines = null;
+
+        try {
+            invLines = findInvalidLines(source);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if(!(invLines.isEmpty())) {   // if there are invalid lines print their number
```

Comment:
You can delete the round brackets as you did below
```suggestion
        if(!invLines.isEmpty()) {   // if there are invalid lines print their number
```